packages/adp-tooling/src/prompts/change-data-source/index.ts

```diff
@@ -0,0 +1,58 @@
+import type { NumberQuestion, ListQuestion, InputQuestion, YUIQuestion } from '@sap-ux/inquirer-common';
+import type { ManifestNamespace } from '@sap-ux/project-access';
+import type { ChangeDataSourceAnswers } from '../../types';
+import { t } from '../../i18n';
+import { filterDataSourcesByType } from '@sap-ux/project-access';
+import { isNotEmptyString } from '../../base/helper';
+
+/**
+ * Gets the prompts for changing the data source.
+ *
+ * @param {Record<string, ManifestNamespace.DataSource>} dataSources - Data sources from the manifest.
+ * @returns {YUIQuestion<ChangeDataSourceAnswers>[]} The questions/prompts.
+ */
+export function getPrompts(
+    dataSources: Record<string, ManifestNamespace.DataSource>
+): YUIQuestion<ChangeDataSourceAnswers>[] {
+    const dataSourceIds = Object.keys(filterDataSourcesByType(dataSources, 'OData'));
+    return [
+        {
+            type: 'list',
+            name: 'id',
+            message: t('prompts.oDataSourceLabel'),
+            choices: dataSourceIds,
+            default: dataSourceIds[0],
+            store: false,
+            guiOptions: {
+                mandatory: true,
+                hint: t('prompts.oDataSourceTooltip')
+            }
+        } as ListQuestion<ChangeDataSourceAnswers>,
+        {
+            type: 'input',
+            name: 'uri',
+            message: t('prompts.oDataSourceURILabel'),
+            guiOptions: {
+                mandatory: true,
+                hint: t('prompts.oDataSourceURITooltip')
+            },
+            validate: isNotEmptyString,
+            when: !!dataSourceIds.length,
+            store: false
+        } as InputQuestion<ChangeDataSourceAnswers>,
+        {
+            type: 'number',
+            name: 'maxAge',
+            message: t('prompts.maxAgeLabel'),
+            when: (answers: ChangeDataSourceAnswers) => answers.uri !== ''
```

Comment:
Why do we need the `maxAge` question? I haven't seen that setting in any UI5 app (ever). Is this something that might have been useful in the past but is obsolete now?

Reply:
Not my expertise but it is described in the documentation as optional:
https://help.sap.com/docs/bas/developing-sap-fiori-app-in-sap-business-application-studio/replacing-odata-service?q=maxAge

Reply:
I'd still remove it because if an app developer never uses it, it is unlikely to me that someone would set it in an adaptation project, so the question is more confusing then helpful, however, I leave that choice to you.

Reply:
I discussed with our PO and we agreed that we would like to keep this questions as optional, to be in sync with the documentation. If this is okay with you?